frappe/utils/identicon.py

```diff
@@ -100,8 +100,6 @@ def base64(self, format="PNG"):
 		if format.upper() not in Image.SAVE:
 			Image.init()
 		save_handler = Image.SAVE[format.upper()]
-		try:
-			save_handler(self.image, fp, "")
-		finally:
-			fp.seek(0)
-			return "data:image/png;base64,{0}".format(base64.b64encode(fp.read()))
+		save_handler(self.image, fp, "")
+		fp.seek(0)
+		return "data:image/png;base64,{0}".format(base64.b64encode(fp.read()))
```

Comment:
This change is not equivalent. `fp.seek` won't be called in case `save_handler` throws and exception without finally.



Everything else, LGTM.